server/types/types.go

```diff
@@ -123,6 +123,19 @@ type DataResponseV1 struct {
 	Result      *interface{} `json:"result,omitempty"`
 }
 
+// DiagnosticsResponseElementV1 models an element in the response message for the
+// Diagnostics API.
+type DiagnosticsResponseElementV1 struct {
+	Query     string `json:"query"`
+	Timestamp int64  `json:"timestamp"`
+
+	Input       interface{} `json:"input,omitempty"`
```

Comment:
Nit: if these are meant to hold any JSON value (including `null`) then they should be pointers (so that `null` can be represented.)